### PR TITLE
ci: implement an optimization for the integration tests

### PR DESCRIPTION
**Pull request by ahal**
This ensures we only schedule the integration tests that are relevant to the PR at hand.

### CODE REVIEW

File: taskcluster/fxci_config_taskgraph/optimizations.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import os
+import subprocess
+from functools import cache
+
+from taskgraph.optimize.base import OptimizationStrategy, register_strategy
+from taskgraph.optimize.strategies import SkipUnlessChanged
+
+from fxci_config_taskgraph.util.integration import FIREFOXCI_ROOT_URL
```
Comment (hneiva) [inline taskcluster/fxci_config_taskgraph/optimizations.py:8]: ❗  This will conflict with #223

File: taskcluster/test/test_optimizations.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+
+import pytest
+from pytest_taskgraph import make_task
+
+from fxci_config_taskgraph import optimizations
+from fxci_config_taskgraph.util.integration import FIREFOXCI_ROOT_URL, STAGING_ROOT_URL
```
Comment (hneiva) [inline taskcluster/test/test_optimizations.py:9]: ❗ This will conflict with #223

File: taskcluster/fxci_config_taskgraph/optimizations.py (hunk 1 above)
Comment (ahal) [reply taskcluster/fxci_config_taskgraph/optimizations.py:8]: Thanks for the heads up. I'm happy to rebase after that lands.

File: taskcluster/fxci_config_taskgraph/optimizations.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        env = os.environ.copy()
+        env["TASKCLUSTER_ROOT_URL"] = FIREFOXCI_ROOT_URL
+        proc = subprocess.run(cmd, capture_output=True, text=True, env=env)
+        lines = [line for line in proc.stdout.splitlines() if line.startswith("!")]
+
+        for i, line in enumerate(lines):
+            line = line.split()[1]
+            assert line.startswith("WorkerPool=")
+            line = line.split("=", 1)[1]
+            lines[i] = line
```
Comment (hneiva) [inline taskcluster/fxci_config_taskgraph/optimizations.py:33]: nit: This feels like a foot gun. A different variable for storing modified worker pools would be cleaner and easier to debug.

**Review by hneiva — APPROVED** [review]
LGTM
just one nit + the conflict

**Pull request merged**